### PR TITLE
Update Getting Started docs to use Swift main branch features

### LINKED ISSUE #126: Update Getting Started docs to use Swift main branch features
We now have functional support for swift-testing in Swift Package Manager (as of https://github.com/apple/swift-package-manager/pull/7047). We should update our documentation:

- Point developers to the main branch instead of 5.10 again;
- Show developers how to invoke `swift test` in a way that triggers swift-testing; and
- Show developers how to use `XCTestScaffold` _only_ when using an older toolchain (with the appropriate caveats and nuances emphasized.)

### PR DESCRIPTION
**Pull request by grynspan**
We now have functional support for swift-testing in Swift Package Manager (as of https://github.com/apple/swift-package-manager/pull/7047). We should update our documentation:

- Point developers to the main branch instead of 5.10 again;
- Show developers how to invoke `swift test` in a way that triggers swift-testing; and
- Show developers how to use `XCTestScaffold` _only_ when using an older toolchain (with the appropriate caveats and nuances emphasized.)

Resolves #126.

### Checklist:

- [x] Code and documentation should follow the style of the [Style Guide](https://github.com/apple/swift-testing/blob/main/Documentation/StyleGuide.md).
- [x] If public symbols are renamed or modified, DocC references should be updated.

### CODE REVIEW

File: Sources/Testing/Testing.docc/TemporaryGettingStarted.md [hunk 1]
```diff
@@ -119,18 +90,57 @@ export TOOLCHAINS=swift
...
 from the command line or click the Product&nbsp;&rarr;&nbsp;Test menu item in
-Xcode. You're done!
+Xcode.
 
 Tests will run embedded in an `XCTestCase`-based test function named
 `testAll()`. If a test fails, `testAll()` will report the failure as its own.
 
+#### Swift 5.9 or earlier
+
+The testing library does not support Swift 5.9 or earlier.
```
Comment (Kyle-Ye) [inline Sources/Testing/Testing.docc/TemporaryGettingStarted.md:142]: If this is the final decision and we do not intend to support Swift 5.9 (Although it compiles fine now with Swift 5.9)

Should we also update `Package.swift`'s version to 5.9?

**Review by Kyle-Ye — COMMENTED** [review]
```
// README.md
We are working to integrate it with the rest of the Swift ecosystem.
```
Maybe we can also update the description here to reflect the latest state directly on README.md

**Comment by grynspan** [comment]
But we're not done working! :)

File: Sources/Testing/Testing.docc/TemporaryGettingStarted.md (hunk 1 above)
Comment (grynspan) [reply Sources/Testing/Testing.docc/TemporaryGettingStarted.md:142]: Well we'd probably update it to 5.11 or later, but we're not quite there yet and we know some folks (including you!) want to continue to experiment.

But yes, we will eventually need to update our package version. Want to file an issue for that?

**Comment by Kyle-Ye** [comment]
> But we're not done working! :)

Got it. I thought the main part is to integrate with SwiftPM. Wondering what's else left.

**Comment by grynspan** [comment]
I think that might be a better topic for the forums than for this PR! 😁

File: Sources/Testing/Testing.docc/TemporaryGettingStarted.md [hunk 2]
```diff
@@ -17,33 +17,20 @@ Start running tests in a new or existing XCTest-based test target.
 
 ## Overview
 
-The testing library is not (yet) integrated into Swift Package Manager, but a
-temporary mechanism is provided for developers who want to start using it with
-their Swift packages.
+The testing library is integrated with Swift Package Manager's `swift test`
```
[the remark this answers is not in the document]
Comment (stmontgomery) [reply Sources/Testing/Testing.docc/TemporaryGettingStarted.md:20]: Maybe this should mention that the SwiftPM integration is in place, although it is still experimental? The command below does include the experimental flag but I'm just thinking that the prose text could explicitly describe the integration as being experimental too

[the remark this answers is not in the document]
Comment (grynspan) [reply Sources/Testing/Testing.docc/TemporaryGettingStarted.md:20]: Sure.

**Comment by grynspan** [comment]
@swift-ci please test

**Pull request merged**